src/dcap_provider.cpp

```diff
@@ -933,10 +964,14 @@ extern "C" quote3_error_t sgx_ql_get_quote_config(
         buf += cert_data_size;
         assert(buf == buf_end);
 
-        time_t expiry;
-        if (get_cache_expiration_time(CollateralTypes::PckCert, expiry))
+        auto retval = convert_to_intel_error(get_cache_header_operation);
+        if (retval == SGX_QL_SUCCESS)

```

Comment:
If you're going to fail based on the get cache header operation failing, why do all the stuff between 940 and 966? Shouldn't the code which retrieves the cache control header be closer to where it's used?

Reply:
Thanks - fixed